pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -264,6 +264,9 @@ type PoolSpec struct {
 	// The device class the OSD should set to (options are: hdd, ssd, or nvme)
 	DeviceClass string `json:"deviceClass"`
 
+	// The inline compression mode in Bluestore OSD to set to (options are: none, passive, aggressive, force)
+	CompressionMode string `json:"compressionMode"`
```

Comment:
New CR settings also need to be added to the [CRD schema](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L469). You should be able to define the enum so only valid options will be input. Note that the schema also needs to be copied to the [helm chart](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/templates/resources.yaml#L389) and [integration tests](https://github.com/rook/rook/blob/master/tests/framework/installer/ceph_manifests.go#L487).

Reply:
Ok, neat! I did have some questions/confusion here.
- Why does the CRD schema not carry the `DeviceClass` or `CrushRoot` properties, although the types.go has them?
- Should these then also be copied to the CephFilesystem [section](https://github.com/rook/rook/blob/9039c674cef0a90cde22e55318535c4799fa3515/cluster/examples/kubernetes/ceph/common.yaml#L292-L312) and by extension the ObjectStore section etc.? As, we could use a CephFS compressed data pool for similar reasons.
- Ack on the helm chart and integration tests needing the change. Do you see an integration test that needs to be added for this PR? I could not come up with one, hence asking.

Reply:
Fixed.

But, I still have the questions and possible additions to other CRD kinds like, CephFilesystem pending.

Reply:
You're correct, those properties are missing from the schema as well, and should be mirrored to the filesystem and object store CRD schemas. This is something recently being cleaned up, so it looks like some pool properties were missed.

Reply:
All related changes are done.

I have not added the remaining missing fields with this PR. I can pick that up once this is merged, if it is still pending.